flash/core/data/data_source.py

```diff
@@ -461,3 +468,13 @@ class TensorDataSource(SequenceDataSource[torch.Tensor]):
 class NumpyDataSource(SequenceDataSource[np.ndarray]):
     """The ``NumpyDataSource`` is a ``SequenceDataSource`` which expects the input to
     :meth:`~flash.core.data.data_source.DataSource.load_data` to be a sequence of ``np.ndarray`` objects."""
+
+class FiftyOneDataSource(SequenceDataSource[SampleCollection]):
```

Comment:
I believe the class signature here should be

```py
class FiftyOneDataSource(Generic[SampleCollection]):
```

The FiftyOne data source takes a `SampleCollection` as input, not a sequence of filepaths, numpy arrays, or anything like that.

They all *output* sequences, but the type structure refers to the *input* data.